Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -258,7 +258,7 @@ std::optional<InteractionRegion> interactionRegionForRenderedRegion(RenderObject
     bool isInlineNonBlock = renderer.isInline() && !renderer.isReplacedOrInlineBlock();
 
     // The parent will get its own InteractionRegion.
-    if (!isOriginalMatch && !isInlineNonBlock)
+    if (!isOriginalMatch && !isInlineNonBlock && !renderer.style().isDisplayTableOrTablePart())
```

Comment:
I think on the web, some people use `display: table` or similar to do clearfix hacks, not sure that affects the code you're writing, but worth noting.